engine/src/stack.cpp

```diff
@@ -1299,17 +1299,70 @@ void MCStack::applyrect(const MCRectangle &nrect)
 		resize(oldrect . width, oldrect . height);
 }
 
+bool MCStack::isdeletable()
+{
+    bool t_deletable = true;
+    
+    if (t_deletable)
+        t_deletable = parent == NULL || scriptdepth != 0 || flags & F_S_CANT_DELETE;
+    
+    // MW-2012-10-26: [[ Bug 9918 ]] If 'cantDelete' is set, then don't delete the stack. Also
+    //   make sure we throw an error.
+    if (t_deletable)
+        t_deletable = !getflag(F_S_CANT_DELETE);
+    
+    if (t_deletable)
+        t_deletable = !(MCdispatcher->gethome() == this);
+    
+    if (t_deletable)
+        t_deletable = !this -> isediting();
+    
+    if (t_deletable && cards != nil)
```

Comment:
This will cause all objects to be checked for delete twice - remember that top-level objects are owned by the stack, so you only need to check the card object itself for deletability.